test/azure/terraform_azure_vm_example_test.go

```diff
@@ -0,0 +1,191 @@
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformAzureVmExample(t *testing.T) {
+	t.Parallel()
+
+	// subscriptionID is overridden by the environment variable "ARM_SUBSCRIPTION_ID"
+	subscriptionID := ""
+	uniquePostfix := random.UniqueId() // "resource" //
+	expectedVmAdminUser := "testadmin"
+	expectedVMSize := compute.VirtualMachineSizeTypes("Standard_B1s")
+	expectedImageSKU := "2019-Datacenter-Core-smalldisk"
+	expectedImageVersion := "latest"
+	expectedDiskType := "Standard_LRS"
+	expectedSubnetAddressRange := "10.0.17.0/24"
+	expectedPrivateIPAddress := "10.0.17.4"
+	expectedManagedDiskCount := 1
+	expectedNicCount := 1
+
+	// Configure Terraform setting up a path to Terraform code.
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located.
+		TerraformDir: "../../examples/azure/terraform-azure-vm-example",
+
+		// Variables to pass to our Terraform code using -var options.
+		Vars: map[string]interface{}{
+			"postfix":          uniquePostfix,
+			"user_name":        expectedVmAdminUser,
+			"vm_size":          string(expectedVMSize),
+			"vm_image_sku":     expectedImageSKU,
+			"vm_image_version": expectedImageVersion,
+			"disk_type":        expectedDiskType,
+			"private_ip":       expectedPrivateIPAddress,
+			"subnet_prefix":    expectedSubnetAddressRange,
+		},
+	}
+
+	// At the end of the test, run `terraform destroy` to clean up any resources that were created.
+	defer terraform.Destroy(t, terraformOptions)
+
+	// Run `terraform init` and `terraform apply`. Fail the test if there are any errors.
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Run `terraform output` to get the values of output variables.
+	resourceGroupName := terraform.Output(t, terraformOptions, "resource_group_name")
+	virtualMachineName := terraform.Output(t, terraformOptions, "vm_name")
+	expectedVNetName := terraform.Output(t, terraformOptions, "virtual_network_name")
+	expectedSubnetName := terraform.Output(t, terraformOptions, "subnet_name")
+	expectedPublicAddressName := terraform.Output(t, terraformOptions, "public_ip_name")
+	expectedNicName := terraform.Output(t, terraformOptions, "network_interface_name")
+	expectedAvsName := terraform.Output(t, terraformOptions, "availability_set_name")
+	expectedOSDiskName := terraform.Output(t, terraformOptions, "os_disk_name")
+	expectedDiskName := terraform.Output(t, terraformOptions, "managed_disk_name")
+
+	// Comment for PR, to be removed:
+	// Please let me know if there are too many tests or alternate examples, happy to reduce the
+	// complexity and amount of code to be maintained. I tried to illustrate different approaches
+	// we have used in various scenarios to illustrate the flexability of Terratest.
```

Comment:
This is actually fine, but consider breaking up each of the following subtests into a separate function to make it easier to read/follow.

Reply:
Refactored.